src/main/java/com/salam/task_manager/controller/TaskController.java

```diff
@@ -0,0 +1,29 @@
+package com.salam.task_manager.controller;
+
+import com.salam.task_manager.dto.TaskDto;
+import com.salam.task_manager.services.TaskService;
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/tasks")
+@AllArgsConstructor
+public class TaskController {
+
+    private final TaskService taskService;
+
+    @PostMapping
+    public ResponseEntity<TaskDto> createTask(@Valid @RequestBody TaskDto taskDto) {
+        UserDetails userDetails = (UserDetails) SecurityContextHolder
+                .getContext().getAuthentication().getPrincipal();
+        String username = userDetails.getUsername();
+        System.out.println(username);
```

Comment:
Why are you doing this here?

Reply:
`UserDetails userDetails = (UserDetails) SecurityContextHolder
                .getContext().getAuthentication().getPrincipal();`
is a common way to get the logged-in user's username in a Spring Security-secured application.
I am using spring security to get the current username without the need to always pass it the parameter